apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,408 @@
+     *
+     * @public
+     * @memberOf BluetoothConnectionManager
+     * @type {Object}
+     */
+    Profiles: {
+      'HFP': 0x111E,
+      'A2DP': 0x110D
+    },
+
+    /**
+     * The address of device that we are trying to connect.
+     *
+     * @public
+     * @memberOf BluetoothConnectionManager
+     * @type {String}
+     */
+    connectingAddress: null,
+
+    /**
+     * The address of device that we are connected.
+     *
+     * @public
+     * @memberOf BluetoothConnectionManager
+     * @type {String}
+     */
+    connectedAddress: null,
+
+    /**
+     * The map is collecting all of the profiles of connected devices.
+     * And hash by device address.
+     *
+     * @private
+     * @memberOf BluetoothConnectionManager
+     * @type {Object}
+     */
+    _connectedDeviceItemsMap: {},
+
+    /**
+     * Default adapter of Bluetooth.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @type {Object BluetoothAdapter}
+     */
+    _defaultAdapter: null,
+
+    /**
+     * Init BluetoothConnectionManager module.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     */
+    _init: function btcm__init() {
+      // Observe 'defaultAdapter' property for reaching default adapter.
+      AdapterManager.observe('defaultAdapter',
+                             this._onDefaultAdapterChanged.bind(this));
+      this._onDefaultAdapterChanged(AdapterManager.defaultAdapter);
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * updating adapter immediately.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothAdapter} newAdapter
+     */
+    _onDefaultAdapterChanged:
+    function btcm__onDefaultAdapterChanged(newAdapter) {
+      Debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+    },
+
+    /**
+     * It provides a convenient function for panel to connect with a device.
+     * And the panel no need to care about having connected device currently.
+     * The method will disconnect current connected device first.
+     * Then, it will start connecting a paired device with the device's adapter.
+     *
+     * @access public
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothDevice} device
+     * @return {Promise}
+     */
+    connect: function btcm_connect(device) {
+      if (!this._defaultAdapter) {
+        return Promise.reject('default adapter is not existed!!');
+      }
+
+      // In code base v1, it always stopDiscovery before connect with a device.
+      // Shall we stopDiscovery here too?
+
+      // Disconnect current connected device first.
+      Debug('connect(): Want to connect device address = ' + device.address +
+            ', name = ' + device.name);
+      return this._disconnectCurrentConnectedDevice().then(() => {
```

Comment:
IMHO it is not required to have a function designed specifically for disconnecting from the current connected device. We can do the work here.